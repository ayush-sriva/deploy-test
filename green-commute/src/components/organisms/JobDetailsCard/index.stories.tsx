import { ComponentMeta, ComponentStory } from '@storybook/react';
import JobDetailsCard from '.';

const description =
  'Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.';
const aboutCompany =
  'High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.';

export default {
  title: 'components/Organisms/JobDetailsCard',
  component: JobDetailsCard,
} as ComponentMeta<typeof JobDetailsCard>;

export const Template: ComponentStory<typeof JobDetailsCard> = (args) => (
  <JobDetailsCard description={description} aboutCompany={aboutCompany} />
);
