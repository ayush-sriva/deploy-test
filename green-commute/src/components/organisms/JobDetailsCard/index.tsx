/**
 * @author Asish Kalintha <asish.sugun@zemosolabs.com>
 */

import { Grid, Paper, styled } from '@mui/material';
import TypographyComponent from '../../atoms/Typography';
import ButtonComponent from '../../atoms/Button';
import responsiveTheme from '../../../theme';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
import IconButtonComponent from '../../molecules/IconButton';

const MainContainer = styled(Paper)({
  width: '23.47rem',
});
const HeaderItem = styled(Grid)({
  color: `${responsiveTheme.palette.greyScale.contrastText}`,
  paddingBottom: '0.5rem',
  paddingLeft: '1.62rem',
  paddingRight: '1.62rem',
});
const InfoItem = styled(Grid)({
  color: `${responsiveTheme.palette.secondary.darker}`,
  paddingBottom: '1.5rem',
  paddingLeft: '1.5rem',
  paddingRight: '1.62rem',
});
const IconContainer = styled(Grid)({
  padding: '0px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  paddinBottom: '2rem',
  height: '4rem',
});
interface JobDescriptionProps {
  description: string;
  aboutCompany: string;
  onClickHandler?: React.MouseEventHandler<HTMLButtonElement>;
}

const JobDetailsCard: React.FC<JobDescriptionProps> = ({ description, aboutCompany, onClickHandler }) => {
  return (
    <MainContainer elevation={0}>
      <Grid container direction='column'>
        <HeaderItem item>
          <TypographyComponent variant={'body1'} children={'Description'} />
        </HeaderItem>
        <InfoItem item>
          <TypographyComponent variant={'body2'} children={description} />
        </InfoItem>
        <HeaderItem item>
          <TypographyComponent variant={'body1'} children={'About Company'} />
        </HeaderItem>
        <InfoItem item>
          <TypographyComponent variant={'body2'} children={aboutCompany} />
        </InfoItem>
        <InfoItem item>
          <TypographyComponent variant={'body2'}>
            Excellent written and oral communication and presentation skills
            <span>
              {' '}
              <ButtonComponent
                children={'SEE MORE'}
                buttonVariant={'text'}
                color={'primary'}
                size={'large'}
                type={'submit'}
                typoVariant={'body1'}
              />{' '}
            </span>
          </TypographyComponent>
        </InfoItem>
        <IconContainer item data-testid='Button1001'>
          <IconButtonComponent
            children={'Green Commute Routes'}
            endIcon={<ArrowForwardIcon htmlColor={responsiveTheme.palette.primary.main} />}
            onClickHandler={onClickHandler}
          />
        </IconContainer>
      </Grid>
    </MainContainer>
  );
};

export default JobDetailsCard;
