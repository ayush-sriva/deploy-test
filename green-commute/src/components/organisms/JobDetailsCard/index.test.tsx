import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import JobDetailsCard from '.';

const description =
  'Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.';
const aboutCompany =
  'High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.';

function renderJobDetailsCard() {
  render(<JobDetailsCard description={description} aboutCompany={aboutCompany} />);
}

describe('CabCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<JobDetailsCard description={description} aboutCompany={aboutCompany} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Text should render correctly', () => {
    renderJobDetailsCard();
    const textElement = screen.getByText(`${description}`);
    expect(textElement).toBeInTheDocument();
  });
  test('Text should render correctly', () => {
    renderJobDetailsCard();
    const textElement = screen.getByText(`${aboutCompany}`);
    expect(textElement).toBeInTheDocument();
  });
  test('Button element rendered correctly', () => {
    renderJobDetailsCard();
    const ButtonElement = screen.getByTestId('Button1001');
    expect(ButtonElement).toBeInTheDocument();
  });
});
