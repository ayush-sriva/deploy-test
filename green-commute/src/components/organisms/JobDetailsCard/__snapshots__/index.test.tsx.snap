// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabCard renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1320j2h-MuiPaper-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-wtzyda-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Description
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-wtzyda-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        About Company
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-63rm72-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Excellent written and oral communication and presentation skills
        <span>
           
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-kpm0tx-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              SEE MORE
            </p>
          </button>
           
        </span>
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-vqb5yb-MuiGrid-root"
      data-testid="Button1001"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1hge7y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Green Commute Routes
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            color="#30A193"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
